### ISSUE TITLE
keyring setup fails

### ISSUE THREAD

**Issue by bobby320** [jaraco/keyring#230]
Keyring setup fails with "no recommended backend was available" I was similare case "Pyinstaller and keyrings.alt" not sure if these are similar. please suggest.
# keyring set Redis-Nagios abc.xyz@klm.com

Password for 'abc.xyz@klm.com' in 'Redis-Nagios':
Traceback (most recent call last):
  File "/usr/local/bin/keyring", line 9, in <module>
    load_entry_point('keyring==9.0', 'console_scripts', 'keyring')()
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/cli.py", line 101, in main
    return cli.run(argv)
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/cli.py", line 64, in run
    set_password(service, username, password)
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/core.py", line 48, in set_password
    _keyring_backend.set_password(service_name, username, password)
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/backends/fail.py", line 18, in get_password
    raise RuntimeError("No recommended backend was available. Install the "
RuntimeError: No recommended backend was available. Install the keyrings.alt package if you want to use the non-recommended backends. See README.rst for details.
#

**Comment by jaraco** [comment]
You say setup fails but the trace back indicates you're attempting to set a password using the command line after the setup/install has completed.

If you're not using PyInstaller, then that ticket doesn't apply.

Did you read the README.rst as indicated by the error message? You need to configure a backend in your environment, and the recommended backend for Linux requires some additional steps to configure.

**Comment by bobby320** [comment]
Thanks for correcting me, I meant keyring setup command is failing.
I installed following dependencies "secretstorage & dbus-python"
so when I tried to "import keyring" in python2.7 prompt, this is error message I see. Please suggest What i am missing here.

```
# /usr/local/bin/python2.7
Python 2.7.10 (default, Jun  9 2016, 15:32:55)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import keyring
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/__init__.py", line 6, in <module>
    from .core import (set_keyring, get_keyring, set_password, get_password,
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/core.py", line 185, in <module>
    init_backend()
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/core.py", line 66, in init_backend
    keyrings = filter(limit, backend.get_all_keyring())
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/util/__init__.py", line 20, in wrapper
    func.always_returns = func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/backend.py", line 179, in get_all_keyring
    exceptions=TypeError))
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/util/__init__.py", line 29, in suppress_exceptions
    for callable in callables:
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/backend.py", line 171, in is_class_viable
    keyring_cls.priority
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/util/properties.py", line 22, in __get__
    return self.fget.__get__(None, owner)()
  File "/usr/local/lib/python2.7/site-packages/keyring-9.0-py2.7.egg/keyring/backends/SecretService.py", line 29, in priority
    bus = secretstorage.dbus_init()
  File "/usr/local/lib/python2.7/site-packages/secretstorage/__init__.py", line 47, in dbus_init
    from dbus.mainloop.glib import DBusGMainLoop
  File "/usr/local/lib/python2.7/site-packages/dbus/mainloop/glib.py", line 29, in <module>
    from _dbus_glib_bindings import DBusGMainLoop, gthreads_init
ImportError: /usr/local/lib/python2.7/site-packages/_dbus_glib_bindings.so: undefined symbol: dbus_connection_setup_with_g_main
>>> exit()
```

Thanks,

**Comment by mitya57** [comment]
@bobby320 Looks like you are not alone, and dbus-python installations from PyPI may be broken (see also https://github.com/jaraco/keyring/issues/231#issuecomment-227890717).

So I will remove it from install_requires in SecretStorage, and we should recommend the users to install dbus-python from distro packages instead.

**Comment by jaraco** [comment]
@mitya57: I suspect it's also necessary, then, to remove it from install requirements in the latest release and remove add another comment in the readme. I'll do that now.

**Comment by jaraco** [comment]
Fixed in c150080e1.
